stdlib/pathlib.pyi

```diff
@@ -188,16 +189,16 @@ class Path(PurePath):
     def expanduser(self: Self) -> Self: ...
     def read_bytes(self) -> bytes: ...
     def read_text(self, encoding: str | None = ..., errors: str | None = ...) -> str: ...
-    def samefile(self, other_path: str | bytes | int | Path) -> bool: ...
-    def write_bytes(self, data: bytes) -> int: ...
+    def samefile(self, other_path: StrPath) -> bool: ...
+    def write_bytes(self, data: ReadableBuffer) -> int: ...
     if sys.version_info >= (3, 10):
         def write_text(
             self, data: str, encoding: str | None = ..., errors: str | None = ..., newline: str | None = ...
         ) -> int: ...
     else:
         def write_text(self, data: str, encoding: str | None = ..., errors: str | None = ...) -> int: ...
-    if sys.version_info >= (3, 8):
-        def link_to(self, target: StrPath | bytes) -> None: ...
+    if (3, 8) <= sys.version_info < (3, 12):
```

Comment:
I don't think type checkers are able to handle complex `sys.version_info` guards like this

```suggestion
    if sys.version_info >= (3, 8) and sys.version_info < (3, 12):
```